### DISCUSSION TITLE
More database types for servers

### DISCUSSION THREAD

**Discussion by Saturn745** [pelican-dev/panel#67] [Ideas]
Currently only MySQL/MariaDB databases can be created for servers. It would be nice to be able to add other types such as Postgres, then when you go to create a new database you can select which type and it will create it.

**Comment by lancepioch** [comment]
Postgres and SQLite databases will be added and maintained, however that's it.

**Reply by Saturn745** [reply]
> Postgres and SQLite databases will be added and maintained, however that's it.

Will this be for the panel itself?

**Comment by lancepioch** [comment]
This isn't a bad idea, but tbh it's gonna be on the back burner!

**Comment by rmartinoscar** [comment]
Should be fairly easy to accomplish just have to [change the driver](https://github.com/pelican-dev/panel/blob/main/app/Extensions/DynamicDatabaseConnection.php#L11) as long as Laravel supports it

**Comment by Army-py** [comment]
Today, PostgreSQL is more powerful than MariaDB / MySQL, both in terms of functionality and performance. And its popularity is growing year on year, in some cases even surpassing that of the two previous DBMSs. So I think that this would be a proposal to take into account when creating different varieties of databases, either for those that will be used by servers created from the panel or when installing it, instead of having to choose between MySQL and SQLite.

**Comment by shiburito** [comment]
Just wanted to add to this it would be nice to have options for redis & mongo in the future as well. Currently maintaining it by just deploying them as servers but it'd be great to have things that need caching or require a nosql database built in!

**Reply by rmartinoscar** [reply]
Redis is a cache not database driver, also 99% of things using redis don't define prefix thus saving everything in the same store.
MongoDB seems more doable although it would add a dependency [@see](https://laravel.com/docs/12.x/mongodb).
